frontend/src/stimulus/controllers/external-links.controller.ts

```diff
@@ -29,61 +29,84 @@
 import { ApplicationController } from 'stimulus-use';
 import { useMutation } from 'stimulus-use';
 
-const BLANK_LINK_QUERY = 'a[target="_blank"]';
+const EXTERNAL_LINK_QUERY = 'a[href^="http"]';
 const BLANK_LINK_DESCRIPTION_ID = 'open-blank-target-link-description';
 
 const isElement = (node:Node):node is Element => node.nodeType === Node.ELEMENT_NODE;
-const isBlankLink = (elem:Element):elem is HTMLAnchorElement => elem.matches(BLANK_LINK_QUERY);
+const isExternalLink = (elem:Element):elem is HTMLAnchorElement => elem.matches(EXTERNAL_LINK_QUERY) && (elem as HTMLAnchorElement).hostname !== window.location.hostname;
```

Comment:
You could split this into two functions:

```suggestion
const isExternalLink = (elem:Element):elem is HTMLAnchorElement => elem.matches(EXTERNAL_LINK_QUERY) && isLinkExternal(elem as HTMLAnchorElement);
const isLinkExternal = (link:HTMLAnchorElement) => link.hostname !== window.location.hostname;
```

Reply:
Update: we should probably be using `window.location.origin` rather than `window.location.hostname` - since we also want to protocol and port into account. `openExternalLinksInNewTab` uses `origin`.

Additionally, we could move from the `URL` parsing from `updateExternalLink` to this method. So something along the lines of:

```suggestion
const isExternalLink = (link:HTMLAnchorElement) => {
  try {
    const linkUrl = new URL(link.href, window.location.origin);
    return linkUrl.origin !== window.location.origin;
  } catch {
    // Do nothing if the url is invalid.
    return false;
  }
}
````